presto-hive/src/main/java/com/facebook/presto/hive/WriteCompletedEvent.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import io.airlift.event.client.EventField;
+import io.airlift.event.client.EventType;
+
+import javax.annotation.concurrent.Immutable;
+
+import java.util.Map;
+
+@Immutable
+@EventType("WriteCompletedEvent")
+public class WriteCompletedEvent
+{
+    private final String queryId;
+    private final String path;
+    private final String schemaName;
+    private final String tableName;
+    private final String partitionName;
+    private final String storageFormat;
+    private final String writerImplementation;
+    private final String prestoVersion;
+    private final String serverAddress;
+    private final String principal;
+    private final String environment;
+    private final Map<String, String> sessionProperties;
+    private final Long bytes;
+    private final long rows;
+
+    public WriteCompletedEvent(
+            String queryId,
+            String path,
+            String schemaName,
+            String tableName,
+            String partitionName,
+            String storageFormat,
+            String writerImplementation,
+            String prestoVersion,
+            String serverAddress,
+            String principal,
+            String environment,
+            Map<String, String> sessionProperties,
+            Long bytes,
+            long rows)
+    {
+        this.queryId = queryId;
```

Comment:
And for the rest, add `@Nullable` (`javax.annotation`) to the getters and parameters